### ISSUE TITLE
How to force knockback to use a particular ViewModel instead of the default?

### ISSUE THREAD

**Issue by trevorlloydelliott** [kmalakoff/knockback#96]
In the past I've never had a problem with specifying polymorphic ViewModel factories for objects which are deserialized as one great big JSON tree.

However I am currently trying to delay load some models separately as part of a standalone Backbone.Collection and kb.collectionObservable.  My structure looks like this:

```
FilterGroups (Backbone.Collection)
- FilterGroup (Backbone.RelationalModel)
  - Filters (Backbone.Collection) // This defines an actual implementation of a filter as a key/value pair, such as "Time":"5:00PM,9:00PM"
    - Filter (Backbone.RelationalModel)
      - ParameterName (string key)
      - Value (string)
  - Parameters (Backbone.Collection)
    - Parameter (Backbone.RelationalModel) // This defines a type of parameter to filter on, such as "Date" or "Time".
```

The idea is that I can have multiple groups of filters each possibly setting a value on the same parameter more than once.  And each Parameter has many abstract properties which define it.  Therefore I should only load the Parameter once and all of my Filters should be able to cross reference them with a navigation property.

So when I render a view for each filter, it has a navigation property that I can bind to all the abstract properties of that parameter, by using bindings such as "Parameter().DisplayName()" or "Parameter().PossibleValues()".

To be clear, **this is already working** up to this point.  backbone-relational is smart enough to know when I manually populate the Backbone.Collection with JSON it sets the appropriate navigation property on the original models, and knockback.js in turn creates view models.

BUT!  I can't seem to get it to create the proper type of view models.  I have implemented a view model factory which works for the kb.collectionObservable I have created (which allows me to organize ALL parameters), but it doesn't work for the navigation property on the Filter.  It always uses the base kb.ViewModel class.

Here is how I defined the relationship:

```
var Filter = Backbone.RelationalModel.extend(
{
    relations:
    [
        {
            type: Backbone.HasOne,
            key: "Parameter",
            keySource: "ParameterName",
            relatedModel: "Parameter",
            includeInJSON: "ParameterName",
            reverseRelation:
            {
                key: "Filters",
                type: Backbone.HasMany,
                includeInJSON: false,
            }
        },
    ],
});
```

And here's how I defined the parameter and its factory:

```
ParameterFactory =
{
    create: function(parameter, options)
    {
        switch (parameter.get("Type"))
        {
            case 0:
                return new StringParameterViewModel(parameter, options);

            case 1:
                return new BooleanParameterViewModel(parameter, options);

            case 2:
                return new ListParameterViewModel(parameter, options);

            case 3:
                return new DateRangeParameterViewModel(parameter, options);

            case 4:
                return new TimeRangeParameterViewModel(parameter, options);

            default:
                throw "Invalid parameter type attribute.";
        }
    }
};

var Parameter = Backbone.RelationalModel.extend(
{
    idAttribute: "ParameterName",
});
```

Here is how I am creating the Backbone.Collection and the kb.collectionObservable:

```
this.ParameterModels = new ParameterCollection();

this.AllParameters = kb.collectionObservable(this.ParameterModels,
{
    factories:
    {
        "models": ParameterFactory,
    },
});
```

Finally, this is how I load the Parameters with AJAX:

```
$.ajax("/api/Parameters",
{
    dataType: "json",
    contentType: "application/json",
    type: "POST",
    context: this,
    success: function(data, textStatus, jqXHR)
    {
        this.ParameterModels.reset(data);
    }
});
```

Additionally, the base view model also specifies the factory to use
So to be clear, the above works perfectly if I browse the `AllParameters` collection.  It creates the proper polymorphic ViewModel.  However, the ViewModel that is created on the navigation property for the Filter is always a `kb.ViewModel`.

I'm confused about all of the possible places to specify a factory.  My FilterGroup itself defines its factories like so:

```
factories:
{
    "Filters.models": FilterFactory,
    "Filters.models.Parameter.model": ParameterFactory,
}
```

And the base class for the Parameter view models also defines a factory like so, but I don't think it has any effect:

```
var ParameterViewModel = kb.ViewModel.extend(
{
    constructor: function(model, options)
    {
        kb.ViewModel.prototype.constructor.call(this, model,
        {
            factories:
            {
                "model": ParameterFactory,
            },

            options: options,
        });

        return this;
    },
});
```

I am not sure why the View Model that knockback creates for the backbone-relational navigation property is not the correct type.  I also don't understand why it's creating a 2nd ViewModel at all (shouldn't it reuse the existing one?).

So to recap, at any time a FilterGroup has a list of all the Parameters which can be filtered on but could have 0 actual filters.  It has a collection of filters, and at some point I add a filter to the list by using a line of code like the following:

`this.model().get("Filters").add(json);`

When this happens a view model for the filter is created and a `kb.ViewModel` for the `Parameter` navigation property is created instead of using my ViewModel factory.
